owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/StatusRequester.kt

```diff
@@ -0,0 +1,135 @@
+        val redirectedToUnsecureLocation: Boolean
+    )
+
+    fun requestAndFollowRedirects(baseLocation: String, client: OwnCloudClient): RequestResult {
+        var currentLocation = baseLocation + OwnCloudClient.STATUS_PATH
+        var redirectedToUnsecureLocation = false
+        var status: Int
+
+        while (true) {
+            val getMethod = getGetMethod(currentLocation)
+
+            status = client.executeHttpMethod(getMethod)
+            val result =
+                if (status.isSuccess()) RemoteOperationResult<OwnCloudVersion>(RemoteOperationResult.ResultCode.OK)
+                else RemoteOperationResult(getMethod)
+
+            if (result.redirectedLocation.isNullOrEmpty() || result.isSuccess) {
+                return RequestResult(getMethod, status, result, redirectedToUnsecureLocation)
+            } else {
+                val nextLocation = updateLocationWithRedirectPath(currentLocation, result.redirectedLocation)
+                redirectedToUnsecureLocation =
+                    checkIfConnectionIsRedirectedToNoneSecure(
+                        redirectedToUnsecureLocation,
+                        currentLocation,
+                        nextLocation
+                    )
+                currentLocation = nextLocation
+            }
+        }
+    }
+
+    private fun Int.isSuccess() = this == HttpConstants.HTTP_OK
+
+    fun handleRequestResult(
+        requestResult: RequestResult,
+        baseUrl: String
+    ): RemoteOperationResult<OwnCloudVersion> {
+        if (!requestResult.status.isSuccess())
+            return RemoteOperationResult(requestResult.getMethod)
+
+        val respJSON = JSONObject(requestResult.getMethod.getResponseBodyAsString() ?: "")
+        if (!respJSON.getBoolean(NODE_INSTALLED))
+            return RemoteOperationResult(RemoteOperationResult.ResultCode.INSTANCE_NOT_CONFIGURED)
+
+        val ocVersion = OwnCloudVersion(respJSON.getString(NODE_VERSION))
+        // the version object will be returned even if the version is invalid, no error code;
+        // every app will decide how to act if (ocVersion.isVersionValid() == false)
+        val result =
+            if (requestResult.redirectedToUnsecureLocation) {
+                RemoteOperationResult<OwnCloudVersion>(RemoteOperationResult.ResultCode.OK_REDIRECT_TO_NON_SECURE_CONNECTION)
+            } else {
+                if (baseUrl.startsWith(HTTPS_SCHEME)) RemoteOperationResult(
+                    RemoteOperationResult.ResultCode.OK_SSL
+                )
+                else RemoteOperationResult(RemoteOperationResult.ResultCode.OK_NO_SSL)
+            }
+        result.data = ocVersion
+        return result
+    }
+
+    companion object {
+        /**
+         * Maximum time to wait for a response from the server when the connection is being tested,
+         * in MILLISECONDs.
+         */
+        private const val TRY_CONNECTION_TIMEOUT: Long = 5000
```

Comment:
This variable could be used in a lot of operations. We could move it to a constants file and use it on each operation that needs it, don't you think?

Reply:
Do we have a central point where we would put these constants?

Reply:
Since it is used by remote operations, we could add it to RemoteOperation file or add it to HttpConstants file where we already have some timeouts: `DEFAULT_DATA_TIMEOUT`

Reply:
I'd let it stay and refactor it later.